gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/util/DefaultTraversalMetrics.java

```diff
@@ -208,14 +208,14 @@ private void handleNestedTraversals(final Traversal.Admin traversal, final Mutab
     private void appendMetrics(final Collection<? extends Metrics> metrics, final StringBuilder sb, final int indent) {
         // Append each StepMetric's row. indexToLabelMap values are ordered by index.
         for (Metrics m : metrics) {
-            String rowName = m.getName();
+            final StringBuilder metricName = new StringBuilder(m.getName());
 
             // Handle indentation
             for (int ii = 0; ii < indent; ii++) {
-                rowName = "  " + rowName;
+                metricName.insert(0, "  ");
             }
```

Comment:
Would it make sense to put this block before the name then it won't have to do inserts?

Reply:
That is the same previous behavior, but using StringBuilder this time.
https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html#insert-int-java.lang.String-

Reply:
Yes, I get that. What I'm suggesting is that instead of having to `insert` data later on thus incurring the cost of array copies, why not put the indentation loop first, then everything after can be `append` only.

Reply:
Do you mean this way?
```java
  private static String padLeft(final String text, final int amountToPad) {
        // not sure why this method needed to exist. stupid string format stuff and commons utilities wouldn't
        // work for some reason in the context this method was used above.
        final StringBuilder newText = new StringBuilder();
        for (int ix = 0; ix < amountToPad; ix++) {
            newText.append(" ");
        }
        newText.append(text);
        return newText.toString();
    }

```

Reply:
Right.

Reply:
Done in both, thank you.